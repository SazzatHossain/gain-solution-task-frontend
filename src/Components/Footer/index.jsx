import React, { useState, useEffect } from "react";


const Footer = () => {
  return (
    <>
      <footer data-testid ="footer-test" className="relative left-0 bottom-0 w-full bg-gray-700 border-gray-200 dark:bg-gray-900">
        <div  className="flex justify-around flex-wrap min-h-[30vh] w-full bg-gray-700 p-4 text-white">
            <div className='w-[300px] text-justify'>
              <p className={'text-xl pt-4'}>Sazzat Hossain</p>
              <p className={'text-sm pt-2 text-gray-400'}>I am a full-stack developer having 3 years experience in Ruby on Rails and
                2 years experience in React JS. I am currently working in for almost 2 years and involve in two
                major projects of the company.</p>
            </div>
          <div className='w-[300px] text-justify'>
              <p className={'text-xl pt-4'}>My Contacts</p>
              <p className={'text-sm pt-2 text-gray-400'}>331/11, TV link road, East Rampura, Dhaka</p>
              <p className={'text-sm pt-2 text-gray-400'}>Email: sazzathossaindipto@gmail.com</p>
              <p className={'text-sm pt-2 text-gray-400'}>Phone No: 01712595391</p>
            </div>
        </div>
        <div  className="flex justify-around items-center flex-wrap w-full bg-gray-700  text-white">
              <p className={'text-sm mb-4'}>Copyrights Â© 2023 SazzatHossain</p>
        </div>
      </footer>
    </>
  );
};

export default Footer;
