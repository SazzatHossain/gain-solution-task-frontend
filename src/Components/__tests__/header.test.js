import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Footer from "../Footer";

afterEach(() => {
  cleanup();
})
test('renders footer component', () => {
  render(<Footer/>);
  const headerElement = screen.getByTestId("footer-test");
  expect(headerElement).toBeInTheDocument();
});
test('renders footer', () => {
  render(<Footer/>);
  const headerElement = screen.getByTestId("footer-test");
  expect(headerElement).toMatchSnapshot();
});


