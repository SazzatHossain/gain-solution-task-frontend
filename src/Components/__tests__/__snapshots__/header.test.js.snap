// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders footer 1`] = `
<footer
  class="relative left-0 bottom-0 w-full bg-gray-800 border-gray-200 dark:bg-gray-900"
  data-testid="footer-test"
>
  <div
    class=" flex items-center justify-between min-h-[30vh] w-full bg-gray-800 p-4"
  />
</footer>
`;
